.github/workflows/lint_code.yml

```diff
@@ -17,6 +17,4 @@ jobs:
       run: pip install flake8
     # Note: flake8 picks up project-wide configuration options from 'setup.cfg' in SCT's root directory
     - name: Lint with flake8
-      # NB: 'master...branch' is syntactic sugar for 'git diff `git merge-base master branch`..branch'
-      # NB: You can exclude the second ref ('branch') and git diff will take the currently checked-out ref
-      run: git diff origin/master... | flake8 --diff
+      run: git diff --name-only --diff-filter=d --merge-base master | grep '\.py$' | xargs --delimiter='\n' --no-run-if-empty flake8
```

Comment:
Dissecting this command:

* [`git diff --merge-base master`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-emgitdiffemltoptionsgt--merge-baseltcommitgt--ltpathgt82308203): Generates diff between working tree and `master`. 
    * (With no other options, this will output the diff [as a patch by default](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---patch), which is very verbose.)
* [`--name-only`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---name-only):  Will suppress the patch output, and display only filenames of changed files.
* [`--diff-filter=d`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203): Display all kinds of changed files (e.g. added, copied, modified, renamed, etc.) **except** files that are deleted (lowercase `d` == exclusion). This is necessary because trying to run `flake8` on a nonexistent file will result in error.
* `| grep '\.py$'`: Filter the output of `git diff` to include only filenames that end in `.py`.
* `| xargs flake8`: Run `flake8` once per filename, using each filename as an argument.
* `--delimiter='\n'`: This is necessary because, by default, `xargs` uses whitespace to delimit input. But, `git diff` outputs each filename on its own line.
* `--no-run-if-empty`: This is necessary because running `flake8` with no arguments will attempt to process the entire working directory. (And, "no arguments" might happen if a PR is opened that only modifies non-Python files.)

This all seems very reasonable to me!

Reply:
A small correction: `| xargs flake8` tries to run `flake8` with many files at once, not once per file.

For example:
```sh
$ seq 10 | xargs echo cmd
cmd 1 2 3 4 5 6 7 8 9 10
$ seq 10 | xargs --max-args=3 echo cmd
cmd 1 2 3
cmd 4 5 6
cmd 7 8 9
cmd 10
```

Reply:
Oops! Thank you for the correction.